src/causalprog/graph/node.py

```diff
@@ -0,0 +1,115 @@
+"""Graph nodes."""
+
+from __future__ import annotations
+
+from abc import abstractproperty
+from typing import Protocol, runtime_checkable
+
+
+class DistributionFamily:
+    """Placeholder class."""
+
+
+class Distribution:
+    """Placeholder class."""
+
+
+@runtime_checkable
+class Node(Protocol):
+    """An abstract node in a graph."""
+
+    @abstractproperty
```

Comment:
Going to X-reference to here <https://github.com/UCL/causalprog/pull/8#discussion_r1993082197>